UrbanCompass/librdkafka
# merge upstream 2022 04 08

- Fix memory leak in admin requests
- Fix MinGW Travis build issues by breaking test execution into a separate script
- ACL Admin Apis: CreateAcls, DescribeAcls, DeleteAcls
- Minor ACL API adjustments and some small code tweaks
- Add ACL support to CHANGELOG
- Retrieve jwt token from token provider (@jliunyu, #3560)
- Fixed typo
- MsgSets with just aborted msgs raised a MSG_SIZE error, and fix backoff (#2993)
- test 0129: style fix
- test 0105: Fix race condition
- Idempotent producer: save state for removed partitions
- Remove incorrect comment on mock API
- Fix rkbuf_rkb assert on malformed JoinGroupResponse.metadata
- clusterid() would fail if there were no topics in metadata (#3620)
- sasl.oauthbearer.extensions should be optional
- Added AK 3.1.0 to test versions
- Changelog updates
- Bump version to v1.9.0
- sasl.oauthbearer.scope should be optional
- Fix hang in list_groups() when cluster is unavailable (#3705)
- Style fixes
- Integration test for OIDC (#3646)
- misc.c style fix
- Test fixes: OIDC requires AK 3.1, not 3.0
- Test 0113: reset security.protocol when using mock cluster
- Travis: use Py 3.8 (not 3.5) on Xenial builder
- Travis: bump integration test from AK 2.7.0 to 2.8.1
- Fix README release wording
- Improve subscribe() error documentation
- Fix linger.ms/message.timeout.ms config checking (#3709)
- Replace deprecated zookeeper flag with bootstrap (@ladislavmacoun, #3700)
- Save one superfluous message timeout toppar scan
- Update to fedora:35 to fix the CentOS 8 build
- Add link to tutorial on Confluent Developer
- Grooming (compilation warnings, potential issues)
- fix: acl binding enum checks (@emasab, #3741)
- cmake: Use CMAKE_INSTALL_LIBDIR
- Trigger op callbacks regardless for unhandled types in consume_batch_queue() et.al. (#3263)
- AppVeyor: Use Visual Studio 2019 image to build since 2015 has TLS problems
- mklove: add LD_LIBRARY_PATH to libcurl builder so that runtime checks pass
- Travis: build alpine & manylinux builds with --source-deps-only
- Nuget Debian build: use --source-deps-only to avoid external dependencies
- RPM test: Use ubi8 image instead of centos:8
- Curl 7.82.0
- mklove: curl now requires CoreFoundation and SystemConfiguration frameworks on osx
- Test 0128: skip if there's no oauthbearer support
- Test 0128: make thread-safe
- Test 0077: reduce flakyness by expediting compaction
- Update to zlib 1.2.12 and OpenSSL 1.1.1n
- vcpkg: revoke to zlib 1.2.11 since 1.2.12 is not yet available (as vcpkg)
- Travis: Disable mingw dynamic build for now (gcc breakage)
- mklove: fix formatting of skipped pkg-config checks
- Fix lock order for rk_init_lock to avoid deadlock (non-released regression)
- vcpkg version bumps
- Update release instructions
- Make dynamic MinGW build copy DLLs instead of trying to manipulate PATH (@neptoess, #3787)
- Fix regression from last PR: curl_ldflags
- Reset stored offset on assign() and prevent offsets_store() for unassigned partitions
- Include broker_id in offset reset logs and corresponding consumer errors (#3785)
- Txn: properly handle PRODUCER_FENCED in InitPid reply
- Provide reason to broker thread wakeups in debug logs
- rdkafka_performance: include broker in DR printouts
- Make SUBTESTS=.. match all of the subtest format string
- Added file io abstraction
- rdkafka_performance: cut down on the number of poll calls in full-rate mode
- Added test.mock.broker.rtt
- Log mock broker bootstrap.servers addresses when test.mock.num.brokers is set
- Mock brokers now allow compressed ProduceRequests
- Made rd_buf_read|peek_iXX() type safe
- SUB_TEST_SKIP() format verification
- Statistics: let broker.wakeups metric cover all broker wakeups, both IO and cnds
- Improved producer queue wakeups
- Broker thread: don't block on IO if there are ops available
- vcpkg: Update to zlib 1.2.12
- Fix some win32 compilation warnings
- Proper use of rd_socket_close() on Win32
- Test 0101: missing return after Test::Skip()
- seek() doc clarification (#3004)
- Documentation updates
- style-check* now fails on style warnings
- Automatic style fixes
- Some OIDC documentation fixes
